### ISSUE TITLE
Use of unannotated unconstrained type parameter in nullable diabled code

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#34842]
The issue is the treatment of a reference to a (unconstrainted) type parameter that was both declared and referenced in a disabled context (without an annotation).  Such a reference should designate an oblivious type and should be treated accordingly.  That does not appear to be the treatment today.  That is the issue in `M1` below.  There should be no problem with the default expression because it is being converted to an oblivious type.

See also https://github.com/dotnet/roslyn/issues/30214

``` c#
#nullable disable
class A<T1, T2, T3> where T2 : class where T3 : object
{
    T1 F1;
    T2 F2;
    T3 F3;
    B F4;

#nullable enable
    void M1()
    {
        F1 = default; // unexpected warning
        F2 = default; // unexpected warning
        F3 = default; // unexpected warning
        F4 = default;
    }
}

class B {}
```
